### PR TITLE
homescreen another items

### PR DESCRIPTION
**Pull request by aragideti**

### CODE REVIEW

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/OneWay.kt [hunk 1]
```diff
@@ -96,9 +103,82 @@ fun OneWay(
                 top.linkTo(parent.top, margin = 64.dp)
             }
         )
+
+        var isDatePickerVisible =  remember { mutableStateOf(false) }
+        var selectedDate = remember { mutableStateOf(Calendar.getInstance()) }
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/OneWay.kt:108]: @aragideti that should not be stored on UI, let's move this to view model

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/OneWay.kt [hunk 2]
```diff
@@ -170,4 +250,62 @@ fun DepartureItem(
...
+@Composable
+fun SmallItemOnlyText(
+    text: Int,
+    fontSize: Dp,
+    modifier: Modifier,
+) {
+    Box(
+        modifier = modifier
+            .fillMaxSize(),
+        contentAlignment = Alignment.Center
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/OneWay.kt:302]: @aragideti I think that there's no reason to wrap this text into `Box` and you can simply use `Row` (not mandatory change, just as a tip)

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    Dialog(onDismissRequest = { onDismissRequest() }) {
+        DatePickerUI(label, onDismissRequest)
+    }
+}
+
+@Composable
+fun DatePickerUI(
+    label: String,
+    onDismissRequest: () -> Unit
+) {
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:48]: @aragideti let's add `modifier` as a parameter here

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.fillMaxWidth(),
+                textAlign = TextAlign.Center
+            )
+
+            Spacer(modifier = Modifier.height(30.dp))
+
+            val chosenYear = remember { mutableStateOf(currentYear) }
+            val chosenMonth = remember { mutableStateOf(currentMonth) }
+            val chosenDay = remember { mutableStateOf(currentDay) }
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:71]: @aragideti you can use `by` here to remove `.value` calls when updating the state

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            items = years,
+            firstIndex = Int.MAX_VALUE / 2 + (currentYear - 1967),
+            onItemSelected = onYearChosen
+        )
+    }
+}
+
+@Composable
+fun InfiniteItemsPicker(
+    modifier: Modifier = Modifier,
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:136]: @aragideti `modifier` should be the first parameter with default value, so place parameters without default value first and then `modifier`

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    items: List<String>,
+    firstIndex: Int,
+    onItemSelected: (String) -> Unit,
+) {
+
+    val listState = rememberLazyListState(firstIndex)
+    val currentValue = remember { mutableStateOf("") }
+
+    LaunchedEffect(key1 = !listState.isScrollInProgress) {
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:145]: @aragideti it's better to use scroll in progress without `!` and check it inside LaunchedEffect

```
LaunchedEffect(key1 = listState.isScrollInProgress) {
    if (!listState.isScrollInProgress) {
        ...
    }
}
```

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        onItemSelected(currentValue.value)
+        listState.animateScrollToItem(index = listState.firstVisibleItemIndex)
+    }
+
+    Box(modifier = Modifier.height(106.dp)) {
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:150]: @aragideti I think you forgot to pass the parent modifier as a Box modifier

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                    val index = it % items.size
+                    if (it == listState.firstVisibleItemIndex + 1) {
+                        currentValue.value = items[index]
+                    }
+
+                    Spacer(modifier = Modifier.height(6.dp))
+
+                    Text(
+                        text = items[index],
+                        modifier = Modifier.alpha(if (it == listState.firstVisibleItemIndex + 1) 1f else 0.3f),
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:165]: @aragideti I think it's better to wrap alpha computation into `derivedStateOf` block

File: app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec"
```
Comment (slothdeboss) [inline app/src/main/java/com/compose/flyticketsbooking/feature/home/tabs/tabScreen/DatePicker.kt:196]: @aragideti let's use `string-array` for this purpose

**Review by igel911 — APPROVED** [review]

**Pull request merged**